Added e2e tests for DR

mmaslankaprv opened the pull request:
Added first e2e cluster linking test.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

mmaslankaprv committed:
admin: added converter for consumer offsets mirroring config

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
tests: added consumer group offsets mirroring test

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv commented:
/dt

mmaslankaprv commented:
/dt

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71595-->
<!--52ec0f6a-71b9-454f-9a62-826ddc2df87e-->
<!--dd5b496b-b8fe-41b2-b063-a05161087cc0-->
<!--title-->
## Retry command for [Build#71595](https://buildkite.com/redpanda/redpanda/builds/71595)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_consumer_groups_mirroring
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_can_not_create_more_than_one_link
```
<!--/body-->

mmaslankaprv commented:
/dt

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71600-->
<!--e0df1645-de5f-4c21-b32d-0a6e7e70d200-->
<!--c7eecbb6-e789-445b-867a-03e2477a0832-->
<!--f5f65537-defa-42a3-8986-e00611405c80-->
<!--title-->
## Retry command for [Build#71600](https://buildkite.com/redpanda/redpanda/builds/71600)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_consumer_groups_mirroring
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_can_not_create_more_than_one_link
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_create_simple_link
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--25ddc3fa-8dca-48bd-b205-020d77502696-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71600">build#71600</a></summary>


| test_class               | test_method                            | test_arguments                                                                                                                          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:-------------------------|:---------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkBasicTests     | test_can_not_create_more_than_one_link | null                                                                                                                                    | integration | https://buildkite.com/redpanda/redpanda/builds/71600#0198f54a-8084-4a22-9788-b75067cb3a6d | FAIL          | 0/1      |                                                                                                                                                           |
| ShadowLinkBasicTests     | test_consumer_groups_mirroring         | null                                                                                                                                    | integration | https://buildkite.com/redpanda/redpanda/builds/71600#0198f54a-8085-4f30-88e9-a70b06f0dfd8 | FAIL          | 0/1      |                                                                                                                                                           |
| ShadowLinkBasicTests     | test_create_simple_link                | null                                                                                                                                    | integration | https://buildkite.com/redpanda/redpanda/builds/71600#0198f54a-8085-4b41-8eaa-c3bbc7aa0293 | FAIL          | 0/1      |                                                                                                                                                           |
| RandomNodeOperationsTest | test_node_operations                   | {"cloud_storage_type": 2, "compaction_mode": "sliding_window", "enable_failures": false, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/71600#0198f547-9013-4b88-86bd-b75d6f0aa20f | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

mmaslankaprv committed:
cl/manager: create consumer offsets topic when link needs it

In empty Redpanda cluster `__consumer_offsets` topic is not initially
present. It is created only when the topic is accessed through Kafka
APIs.

In order for migration of consumer groups to work we require topic to be
present as it controls where the consumer group migration task is
running.

Added piece of code that creates the topic whenever a link with consumer
group migrations is created.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

michael-redpanda commented on src/v/redpanda/admin/services/shadow_link/converter.cc:
```diff
@@ -106,6 +106,20 @@ create_topic_metadata_mirroring_config(
+  const proto::admin::consumer_offset_sync_options& options) {
+    cluster_link::model::consumer_groups_mirroring_config config;
+
+    if (options.get_interval() > absl::ZeroDuration()) {
+        config.task_interval = absl::ToChronoNanoseconds(
+          options.get_interval());
+    }
```
```suggestion
    if (options.interval.has_value()) {
        config.task_interval = absl::ToChronoNanoseconds(
          options.get_interval());
    }
```

mmaslankaprv replied:
I used the same pattern as used for topic metadata config, which one is correct ?

michael-redpanda replied:
oh crap my bad, this is PB -> model
